modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -39,9 +39,15 @@ import zio.json.ast.Json
 import zio.json.ast.Json.{Arr, Obj}
 import zio.schema.Schema
 
-sealed trait BulkableRequest[A] extends ElasticRequest[A]
+private[elasticsearch] sealed trait IsExecutable
+private[elasticsearch] sealed trait Executable    extends IsExecutable
+private[elasticsearch] sealed trait NotExecutable extends IsExecutable
```

Comment:
Move this outside of `ElasticRequest`. I guess you can put it into `request` package.